pyglmnet/pyglmnet.py

```diff
@@ -139,6 +139,10 @@ def _logL(distr, y, y_hat, z=None):
         # https://www.statistics.ma.tum.de/fileadmin/w00bdb/www/czado/lec8.pdf
         nu = 1.  # shape parameter, exponential for now
         logL = np.sum(nu * (-y / y_hat - np.log(y_hat)))
+    elif distr == 'neg-binomial':
+        r = 15.
```

Comment:
- let's define this a class param: say `neg_binomial_n_failures` (or `neg_binomial_n_successes`) with a prespecified default
- why `r` here and `theta` in cheatsheet?

Reply:
> * let's define this a class param: say `neg_binomial_n_failures` (or `neg_binomial_n_successes`) with a prespecified default

Make sense to me. It would be good if the users were able also to play with this parameter
(like by specifying it when constructing the GLM object). 

> * why `r` here and `theta` in cheatsheet?

mmmh probably a typo from my side.

Reply:
yes, the parameter `neg_binomial_n_failures` should be exposed to the user while constructing the object

Reply:
I think the class will start getting bloated if we expose the parameters of each new distribution. What many libraries do is to allow passing an object of a class so you can use that to pass whatever parameters are specific to the distribution. E.g.,

```py
distr = NegativeBinomial(n_failure)
GLM(distr=distr)
```

that way you can also allow arbitrary distributions:

```py
class MyDistribution(BaseDistribution):
    def loglikelihood(self):
        pass
    def loglik_gradient(self):
       pass
   def loglik_hessian(self):
       pass
```

Reply:
@jasmainak totally agree and acknowledge your similar comment in the thread above. 

i worry though that this is a bit out of scope for the current PR and will also result in breaking changes to our API and docs. once we get the negative binomial distribution right in this PR, we can make a separate PR to refactor into distribution-specific classes.

Reply:
Agreed! and I had forgotten about the comment above.

I think this PR is closed to being merged once it's cleaned up?